src/CMakeLists.txt

```diff
@@ -341,6 +341,48 @@ set(SOURCE_FILES
     WrapCalls.cpp
     )
 
+##
+# Build time tools
+##
+
+if (CMAKE_CROSSCOMPILING)
+    # Note: using CMAKE_EXECUTABLE_SUFFIX breaks cross-OS builds involving Windows.
+    include(ExternalProject)
+    ExternalProject_Add(host-tools
```

Comment:
super-duper-nit: I find it hard to quickly read the keywords-vs-arg-values here; I wonder if it would be more obvious with indentation like:

```
    ExternalProject_Add(host-tools
                        BUILD_BYPRODUCTS
                              "<INSTALL_DIR>/bin/build_halide_h${CMAKE_EXECUTABLE_SUFFIX}"
                              "<INSTALL_DIR>/bin/binary2cpp${CMAKE_EXECUTABLE_SUFFIX}"
                        SOURCE_DIR 
                              ${CMAKE_CURRENT_LIST_DIR}/tools
                        INSTALL_DIR 
                              ${CMAKE_CURRENT_BINARY_DIR}/host-tools
                        CMAKE_ARGS
                              -D CMAKE_TOOLCHAIN_FILE=${Halide_HOST_TOOLCHAIN_FILE}
                              -D CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
                              -D CMAKE_INSTALL_PREFIX=<INSTALL_DIR>
                        BUILD_ALWAYS 
                              YES)
```

Reply:
The formatter in CLion is not very smart and it tramples such indentation :/ I could add some blank lines to make it a bit more readable.

Reply:
Didn't know there existed an autoformatter for CMake. Does it exist anywhere other than CLion?

Reply:
Not that I'm aware of. There are two open-source CMake formatters I'm aware of: [cmake-format](https://github.com/cheshirekow/cmake_format) (I tried it and it does a bad job, the linter is even worse) and [gersemi](https://github.com/BlankSpruce/gersemi) (not very configurable, led to a huge diff when I tried it, and pre-alpha)